starter-code/memory.js

```diff
@@ -33,14 +33,81 @@ var MemoryGame = function() {
     this.correctPairs = 0;
 };
 
+MemoryGame.prototype._shuffleCards = function () {
+  // Modern Fisher-Yates shuffling algorithm
+  var length = this.cards.length
+  for (var i = 0; i < length - 2; i++) {
+    var j = Math.floor(Math.random() * (length - i)) + i
+    this.cards = shuffle(i, j, this.cards)
+  }
+  // console.log(this.cards)
+}
+
+function shuffle (IndexCardA, IndexCardB, array) {
+  // Copies the original array and returns a new one with the two items exchanging place
+  var new_array = array.slice()
+  var cardB = array[IndexCardB]
+
+  new_array[IndexCardB] = array[IndexCardA]
+  new_array[IndexCardA] = cardB
+  return new_array
+}
+
+MemoryGame.prototype.selectCard = function (card) {
+  if (this.selectedCards.length === 0) {
+    // Case: Adding a first card
+    this.selectedCards.push(card)
+  }
+  else {
+    // Case: Adding a second card
+    this.selectedCards.push(card)
+    this.pairsClicked += 1
+    var cardA = this.selectedCards[0]
+    var cardB = this.selectedCards[1]
+
+    if (cardA === cardB) {
+      // Case: Good guess
+      this.selectedCards = [] // resetting selectedCards
+      this.correctPairs += 1
+
+      // Updating game score-board on screen
+      $('#pairs_guessed').text(this.correctPairs)
+      $('#pairs_clicked').text(this.pairsClicked)
+
+      // Hiding the front of the two cards, and displaying the backs
+      $('.toUndisplay[name="' + cardA + '"]').removeClass('toUndisplay')
```

Comment:
This is a good idea in my opinion. Seperating a more "logical" thing like setting a property, and the actual class remove with JQuery

Reply:
When we learn VueJS you will see a easier way to work this way, "data-oriented"